File: benchmark/python/mean_variance.py
import time
arr = [i + 1 for i in range(1000000)]
start = time.time() * 1000
s = 0
for x in arr:
    s += x
mean = s / len(arr)
sq_diff = 0
for x in arr:
    d = x - mean
    sq_diff += d * d
variance = sq_diff / len(arr)
end = time.time() * 1000
print(f"Mean: {mean}")
print(f"Variance: {variance}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/nodejs/mean_variance.js
const start = Date.now();
const arr = [];
for (let i = 1; i <= 1000000; i++) arr.push(i);
let s = 0;
for (let i = 0; i < arr.length; i++) s += arr[i];
const mean = s / arr.length;
let sq_diff = 0;
for (let i = 0; i < arr.length; i++) {
    const diff = arr[i] - mean;
    sq_diff += diff * diff;
}
const variance = sq_diff / arr.length;
const end = Date.now();
console.log(`Mean: ${mean}`);
console.log(`Variance: ${variance}`);
console.log(`Time: ${end - start}ms`);

